teachertool/src/services/backendRequests.ts

```diff
@@ -99,19 +100,34 @@ export async function askCopilotQuestionAsync(shareId: string, question: string)
     const data = { id: shareId, question };
     let result: string = "";
     try {
+        const headers = await getAuthHeadersAsync();
+        headers["Content-Type"] = "application/json";
+
         const request = await fetch(url, {
             method: "POST",
-            headers: { "Content-Type": "application/json" },
+            headers,
             body: JSON.stringify(data),
+            credentials: "include",
         });
 
         if (!request.ok) {
-            throw new Error("Unable to reach Copilot");
+            throw new Error("Unable to reach AI");
         }
         result = await request.json();
     } catch (e) {
-        logError(ErrorCode.askCopilotQuestion, e);
+        logError(ErrorCode.askAIQuestion, e);
     }
 
     return result;
 }
+
+async function getAuthHeadersAsync(): Promise<pxt.Map<string>> {
```

Comment:
To avoid code duplication, let's move `getAuthHeadersAsync` to pxtlib/auth.ts. Also update the method `staticApiAsync` in pxtlib/auth.ts to call this function.